### PR TITLE
Add Holiday for Cyprus

### PR DESCRIPTION
**Pull request by bkintanar**
Fixes: https://github.com/azuyalabs/yasumi/issues/284

### CODE REVIEW

**Comment by stelgenhof** [comment]
@bkintanar I think so far it looks good (the tests are green :+1: ) Are these all of the national holidays of Cyprus?

**Comment by bkintanar** [comment]
@stelgenhof no. easterTuesday is a bank holiday. Should it be removed?

**Comment by stelgenhof** [comment]
@bkintanar It can stay, as long as it has the appropriate category. Generally, for each country, all of the official, national holidays should be included in a provider. Any other holidays can of course be added as well.

**Comment by stelgenhof** [comment]
@bkintanar Looking at your PR once more, I noticed that (from the list on https://en.wikipedia.org/wiki/Public_holidays_in_Cyprus) Easter Sunday seems to be missing. Also a number of unit tests for some holidays are not present. Please add unit tests for all the holidays defined by the provider.

File: src/Yasumi/Provider/Cyprus.php [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+     */
+    public function initialize(): void
+    {
+        $this->timezone = 'Asia/Nicosia';
+
+        // Add common holidays
+        $this->addHoliday($this->newYearsDay($this->year, $this->timezone, $this->locale));
+
+        // Add Christian holidays
+        $this->addHoliday($this->epiphany($this->year, $this->timezone, $this->locale));
```
Comment (stelgenhof) [inline src/Yasumi/Provider/Cyprus.php:55]: Please add unit test for this holiday.

File: src/Yasumi/Provider/Cyprus.php [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+        $this->addHoliday($this->christmasEve($this->year, $this->timezone, $this->locale));
```
Comment (stelgenhof) [inline src/Yasumi/Provider/Cyprus.php:56]: Please add unit test for this holiday.

File: src/Yasumi/Provider/Cyprus.php [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+        $this->addHoliday($this->christmasDay($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->secondChristmasDay($this->year, $this->timezone, $this->locale));
```
Comment (stelgenhof) [inline src/Yasumi/Provider/Cyprus.php:58]: Please add unit test for this holiday.

File: src/Yasumi/Provider/Cyprus.php [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+
+        $this->addHoliday($this->goodFriday($this->year, $this->timezone, $this->locale));
```
Comment (stelgenhof) [inline src/Yasumi/Provider/Cyprus.php:60]: Please add unit test for this holiday.

File: src/Yasumi/Provider/Cyprus.php [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+        $this->addHoliday($this->easterMonday($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->easterTuesday($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->pentecostMonday($this->year, $this->timezone, $this->locale));
```
Comment (stelgenhof) [inline src/Yasumi/Provider/Cyprus.php:63]: Please add unit test for this holiday.

File: src/Yasumi/Provider/Cyprus.php [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+
+        // Calculate other holidays
+        $this->calculateGreenMonday();
+        $this->calculateGreekIndependenceDay();
+        $this->calculateGreekCypriotNationalDay();
+        $this->calculateLabourDay();
+        $this->calculateDormitionOfTheTheotokos();
+        $this->calculateCypriotIndependenceDay();
+        $this->calculateGreekNationalDay();
```
Comment (stelgenhof) [inline src/Yasumi/Provider/Cyprus.php:72]: Please add unit test for these holidays.

**Review by stelgenhof — CHANGES_REQUESTED** [review]
Looking at your PR once more, I noticed that (from the list on https://en.wikipedia.org/wiki/Public_holidays_in_Cyprus) Easter Sunday seems to be missing. Also a number of unit tests for some holidays are not present. Please add unit tests for all the holidays defined by the provider.

**Pull request closed without merging**